modules/openapi-generator/src/main/resources/Java/libraries/jersey2/anyof_model.mustache

```diff
@@ -33,9 +37,23 @@ public class {{classname}} extends AbstractOpenApiSchema{{#vendorExtensions.x-im
             JsonNode tree = jp.readValueAsTree();
 
             Object deserialized = null;
+            {{#discriminator}}
+            Class cls = JSON.getClassForElement(tree, {{classname}}.class);
+            if (cls != null) {
+                // When the OAS schema includes a discriminator, use the discriminator value to
+                // discriminate the anyOf schemas.
+                // Get the discriminator mapping value to get the class.
+                // TODO: custom deserialization when additionalProperties is set.
+                deserialized = tree.traverse(jp.getCodec()).readValueAs(cls);
+                {{classname}} ret = new {{classname}}();
+                ret.setActualInstance(deserialized);
+                return ret;
+            }
+            {{/discriminator}}
             {{#anyOf}}
             // deserialzie {{{.}}}
             try {
+                // TODO: custom deserialization when additionalProperties is set.
```

Comment:
@wing328 , here we need to deserialize into the class, but we need to ignore the unknown properties and put them into the `additionalProperties` map.

Reply:
👌 Let me try to come up with a way to handle this use case.